pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -0,0 +1,246 @@
+			Protocol:    &epp.Protocol,
+			AppProtocol: epp.AppProtocol,
+		})
+	}
+	return epsPorts
+}
+
+// boolPtrChanged returns true if a set of bool pointers have different values.
+func boolPtrChanged(ptr1, ptr2 *bool) bool {
+	if (ptr1 == nil) != (ptr2 == nil) {
+		return true
+	}
+	if ptr1 != nil && ptr2 != nil && *ptr1 != *ptr2 {
+		return true
+	}
+	return false
+}
+
+// objectRefPtrChanged returns true if a set of object ref pointers have
+// different values.
+func objectRefPtrChanged(ref1, ref2 *corev1.ObjectReference) bool {
+	if (ref1 == nil) != (ref2 == nil) {
+		return true
+	}
+	if ref1 != nil && ref2 != nil && !apiequality.Semantic.DeepEqual(*ref1, *ref2) {
+		return true
+	}
+	return false
+}
+
+// getEndpointsFromDeleteAction parses an Endpoints resource from a delete
+// action.
+func getEndpointsFromDeleteAction(obj interface{}) *corev1.Endpoints {
+	if endpointSlice, ok := obj.(*corev1.Endpoints); ok {
+		return endpointSlice
+	}
+	// If we reached here it means the EndpointSlice was deleted but its final
+	// state is unrecorded.
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+		return nil
+	}
+	endpoints, ok := tombstone.Obj.(*corev1.Endpoints)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not an Endpoints resource: %#v", obj))
+		return nil
+	}
+	return endpoints
+}
+
+// getEndpointSliceFromDeleteAction parses an EndpointSlice from a delete action.
+func getEndpointSliceFromDeleteAction(obj interface{}) *discovery.EndpointSlice {
+	if endpointSlice, ok := obj.(*discovery.EndpointSlice); ok {
+		return endpointSlice
+	}
+	// If we reached here it means the EndpointSlice was deleted but its final
+	// state is unrecorded.
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+		return nil
+	}
+	endpointSlice, ok := tombstone.Obj.(*discovery.EndpointSlice)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not an EndpointSlice resource: %#v", obj))
+		return nil
+	}
+	return endpointSlice
+}
+
+// endpointsControllerKey returns a controller key for an Endpoints resource but
+// derived from an EndpointSlice.
+func endpointsControllerKey(endpointSlice *discovery.EndpointSlice) (string, error) {
+	if endpointSlice == nil {
+		return "", fmt.Errorf("nil EndpointSlice passed to serviceControllerKey()")
+	}
+	serviceName, ok := endpointSlice.Labels[discovery.LabelServiceName]
+	if !ok || serviceName == "" {
+		return "", fmt.Errorf("EndpointSlice missing %s label", discovery.LabelServiceName)
+	}
+	return fmt.Sprintf("%s/%s", endpointSlice.Namespace, serviceName), nil
+}
+
+// skipMirror return true if the the LabelSkipMirror label has been set to
+// "true".
+func skipMirror(labels map[string]string) bool {
+	skipMirror, _ := labels[discovery.LabelSkipMirror]
```

Comment:
you could just do

return skipMirror == "true"

Reply:
That is a lot cleaner, thanks!